botEngine.js

```diff
@@ -81,8 +81,18 @@ async function listenToMessages(client) {
         authorBuffer.push(createAuthorEntry(message));
         try {
           const response = await fn(message);
+          let isUndefined = false;
```

Comment:
Why are these changes being made?

Reply:
When live testing on my own discord server, I was running into these issues: 

**Error: Request failed with status code 401**
    at addPointsToUser (/Users/timothyling/Desktop/OdinProject/odin-bot-v2/botCommands/points.js:48:11)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /Users/timothyling/Desktop/OdinProject/odin-bot-v2/botCommands/points.js:151:30
    at async Promise.all (index 0)
    at async /Users/timothyling/Desktop/OdinProject/odin-bot-v2/botEngine.js:83:28
[ undefined ]
(node:38481) **UnhandledPromiseRejectionWarning: DiscordAPIError: Cannot send an empty message**
    at RequestHandler.execute (/Users/timothyling/Desktop/OdinProject/odin-bot-v2/node_modules/discord.js/src/rest/RequestHandler.js:154:13)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async RequestHandler.push (/Users/timothyling/Desktop/OdinProject/odin-bot-v2/node_modules/discord.js/src/rest/RequestHandler.js:39:14)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:38481) **UnhandledPromiseRejectionWarning: Unhandled promise rejection.** This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:38481) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

This is happening because I don't have the authorization token to make API calls to the API endpoint when trying to add points to a user's total. Therefore, the axios call within the addPointsToUser function is always going to throw an error. 

`return Promise.all(userIds.map(async (userId, i) => {
try {
          const pointsUser = await addPointsToUser(user.id, userId[1]);
} catch (err) {
          console.log(err);
})`

The Promise.all resolves to an array of the input Promises, which since addPointsToUser threw an error, is outputting [undefined] or [undefined,undefined,undefined] if you used a command that mentioned say 3 users. 

In botEngine.js, that array of undefined is getting passed to message.channel.send, which discord.js is regarding as an empty message, which is one of the errors mentioned above. 

`const response = await fn(message);
if (response) {  
            try {
              message.channel.send(response);
            } catch (e) {
              console.log(e);
            }
}
`
After looking into the issue more, I think my implementation is incorrect. In 

`return Promise.all(userIds.map(async (userId, i) => {
try {
          const pointsUser = await addPointsToUser(user.id, userId[1]);
} catch (err) {
          console.log(err);
})`

I need to return a Promise.reject(err) like so: 
`try {
          const pointsUser = await addPointsToUser(user.id, userId[1]);
} catch (err) {
          console.log(err);
return Promise.reject(err);
}`

Reply:
> This is happening because I don't have the authorization token to make API calls to the API endpoint when trying to add points to a user's total. Therefore, the axios call within the addPointsToUser function is always going to throw an error.

Well, if it's happening because you don't have API access, we don't need to change it here.

Reply:
Disregard the above actually, it's good to have a failsafe if the API returns a bad response.

Reply:
Understood. We can ignore the 401 error, but I think the other two warnings are still valid on the off chance that our backend API goes down or returns any sort of server-side errors.

After doing some more digging, both warnings go away when I explicitly return a Promise.reject(err) in the catch of the userIds.map. When the resolved Promise.all array goes to botEngine.js, the error gets caught there as well, avoiding the logic that was causing the **UnhandledPromiseRejectionWarning: DiscordAPIError: Cannot send an empty message** warning. 


`return Promise.all(
      userIds.map(async (userId, i) => {

        try {
          const pointsUser = await addPointsToUser(user.id, userId[1]);
        } catch (err) {
          console.log(err);
          return Promise.reject(err);
        }
      }),
    );`

Reply:
Agree- I misinterpreted the above.